pennylane/ops/qubit.py

```diff
@@ -533,6 +533,53 @@ def _controlled(self, wire):
         CSWAP(wires=wire + self.wires)
 
 
+class ISWAP(Operation):
+    r"""ISWAP(wires)
+    The i-swap operator
+
+    .. math:: ISWAP = \begin{bmatrix}
+            1 & 0 & 0 & 0 \\
+            0 & 0 & i & 0\\
+            0 & i & 0 & 0\\
+            0 & 0 & 0 & 1
+        \end{bmatrix}.
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int]): the wires the operation acts on
+    """
+    num_params = 0
+    num_wires = 2
+    par_domain = None
+
+    @classmethod
+    def _matrix(cls, *params):
+        return np.array([[1, 0, 0, 0], [0, 0, 1j, 0], [0, 1j, 0, 0], [0, 0, 0, 1]])
+
+    @classmethod
+    def _eigvals(cls, *params):
+        return np.array([1j, -1j])
```

Comment:
Adding a test case in the `class TestOperations` class for verifying the eigenvalues could help.

The test could go something like
```python
def test_eigvals_iswap(self):
    # 1. Create the operation
    # 2. Compute the eigenvalues of op using its matrix representation,
    # e.g., by calling np.linalg.eigh(operation.matrix)
    # 3. Compare the results with the stored eigenvalues
```